pkg/controller/instance/controller_reconcile.go

```diff
@@ -446,16 +426,26 @@ func (igr *instanceGraphReconciler) setManaged(ctx context.Context, obj *unstruc
 
 	igr.log.V(1).Info("Setting managed state", "name", obj.GetName(), "namespace", obj.GetNamespace())
 
-	copy := obj.DeepCopy()
-	if err := metadata.SetInstanceFinalizerUnstructured(copy); err != nil {
+	instancePatch := &unstructured.Unstructured{}
+	instancePatch.SetUnstructuredContent(map[string]interface{}{
+		"apiVersion": obj.GetAPIVersion(),
+		"kind":       obj.GetKind(),
+		"metadata": map[string]interface{}{
+			"name":       obj.GetName(),
+			"namespace":  obj.GetNamespace(),
+			"labels":     obj.GetLabels(),
+			"finalizers": obj.GetFinalizers(),
+		},
+	})
+	if err := metadata.SetInstanceFinalizerUnstructured(instancePatch); err != nil {
 		return nil, fmt.Errorf("failed to set finalizer: %w", err)
 	}
 
-	igr.instanceLabeler.ApplyLabels(copy)
+	igr.instanceLabeler.ApplyLabels(instancePatch)
 
 	updated, err := igr.client.Resource(igr.gvr).
 		Namespace(obj.GetNamespace()).
-		Update(ctx, copy, metav1.UpdateOptions{})
+		Apply(ctx, instancePatch.GetName(), instancePatch, metav1.ApplyOptions{FieldManager: FieldManagerForLabeler, Force: false})
```

Comment:
a force false would lead to errors if we have other field managers on this. Since we are in finalize Deletion, is forcing here a valid case?

Reply:
true. i fail to recollect why i had false here specifically. reverting to true. If we do indeed discover why we need a false later we can add a comment and revert it. For now true makes sense for the force flag.